config: !file tests/http/config/graphql-datasource-errors.graphql
name: Graphql datasource
mock:
  - request:
      url: http://upstream/graphql
      method: POST
      body: '{ "query": "query { user(id: 1) { name } }" }'
    response:
      body:
        data: null
        errors:
          - message: Failed to resolve user
            path: ["user"]
            locations:
              - line: 1
                column: 9
  - request:
      url: http://upstream/graphql
      method: POST
      body: '{ "query": "query { user(id: 2) { name id } }" }'
    response:
      body:
        data:
          user:
            name: null
            id: 2
        errors:
          - message: Failed to resolve name
            locations:
              - line: 1
                column: 35
assert:
  - request:
      method: POST
      url: http://localhost:8080/graphql
      body:
        query: "query { user(id: 1) { name } }"
    response:
      body:
        data:
          user:
            name: null
        errors:
          - message: Failed to resolve user
            path: ["user"]
            locations:
              - line: 1
                column: 9
  - request:
      method: POST
      url: http://localhost:8080/graphql
      body:
        query: "query { user(id: 2) { name id } }"
    response:
      body:
        data:
          user:
            name: null
            id: 2
        errors:
          - message: Failed to resolve name
            locations:
              - line: 1
                column: 35
